pkg/manifests/amcfg.go

```diff
@@ -180,6 +183,16 @@ func ConvertToThanosAlertmanagerConfiguration(ta []AdditionalAlertmanagerConfig)
 
 		cfg.StaticConfigs = a.StaticConfigs
 
+		switch cfg.Scheme {
```

Comment:
ThanosRuler only supports the proxy URL config but CMO can take into account the NoProxy string to decide what is the correct value (HTTP Proxy, HTTPS Proxy or empty string). 
I'd use https://pkg.go.dev/golang.org/x/net/http/httpproxy#Config.ProxyFunc for this.

Reply:
done